chia/wallet/trading/trade_store.py

```diff
@@ -236,6 +301,25 @@ async def get_trade_record_with_status(self, status: TradeStatus) -> List[TradeR
 
         return records
 
+    async def get_coins_of_interest_with_trade_statuses(self, trade_statuses: List[TradeStatus]) -> List[Coin]:
+        """
+        Checks DB for TradeRecord with id: id and returns it.
+        """
+        async with self.db_wrapper.reader_no_transaction() as conn:
+            cursor = await conn.execute(
+                "SELECT distinct c.coin from coin_of_interest_to_trade_record c, trade_records t "
+                "WHERE t.status in (%s) AND c.trade_id = t.trade_id" % ",".join("?" * len(trade_statuses)),
```

Comment:
generally I think we need to be moving away from building SQL query strings in code (because it's risky and may defeat the compiler query cache in python's sqlite binding). However, the pattern of using `in (?,?,...)` with a variable number of arguments is hard to get away from (afaik). However, we seem to have a convention to build the list of question marks with an f-string in other places. It might make sense to conform to that just for consistency.

i.e.

```
f"WHERE t.status in ({','.join('?' * len(trade_statuses))}) AND c.trade_id = t.trade_id"
```

Come to think of it, we should probably factor out this pattern into a utility function, since we use it in quite a few places. (out of scope for you patch clearly)